cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/resource/AzureResourceIdProviderService.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.cloudbreak.cloud.azure.resource;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import com.sequenceiq.cloudbreak.cloud.azure.AzureStorage;
+
+@Component
+public class AzureResourceIdProviderService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AzureResourceIdProviderService.class);
+
+    public String generateImageId(String subscriptionId, String resourceGroup, String imageName) {
+        Assert.hasText(subscriptionId, "Subscription id must not be null or empty.");
+        Assert.hasText(resourceGroup, "Resource group must not be null or empty.");
+        Assert.hasText(imageName, "Image name must not be null or empty.");
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(generateCommonPart(subscriptionId, resourceGroup));
+        sb.append("/providers/Microsoft.Compute/");
+        sb.append(AzureStorage.IMAGES_CONTAINER);
+        sb.append("/");
+        sb.append(imageName);
+        String resourceReference = sb.toString();
+        LOGGER.info("Generated resourceReferenceId: {}", resourceReference);
+        return resourceReference;
+    }
+
+    public String generateDnsZoneDeploymentId(String subscriptionId, String resourceGroup, String deploymentName) {
+        Assert.hasText(subscriptionId, "Subscription id must not be null or empty.");
+        Assert.hasText(resourceGroup, "Resource group must not be null or empty.");
+        Assert.hasText(deploymentName, "Deployment name must not be null or empty.");
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(generateCommonPart(subscriptionId, resourceGroup));
+        sb.append("/providers/Microsoft.Resources/");
+        sb.append("deployments/");
+        sb.append(deploymentName);
+        String resourceReference = sb.toString();
+        LOGGER.info("Generated resourceReferenceId: {}", resourceReference);
+        return resourceReference;
+    }
+
+    public String generateCommonPart(String subscriptionId, String resourceGroup) {
```

Comment:
this could be private, or if we keep this public we might want to add asserts for the parameters also here